crates/tx/src/section.rs

```diff
@@ -61,6 +61,15 @@ pub enum Section {
     MaspBuilder(MaspBuilder),
     /// Wrap a header with a section for the purposes of computing hashes
     Header(Header),
+    /// The account paying the fee for a shielding transaction
+    ShieldingFee {
```

Comment:
Quick idea I haven't reasoned too deeply about (especially in terms of safety): would it be possible to avoid this new section type and encode all the necessary data directly inside the `Transfer` data section? Maybe we could just add two entires: the fee payer as  a source and the PGF account as the target. The `multitransfer` function will just update the balances accordingly. The masp vp should have all the necessary data to check if the PGF account has been credited the fees when the vector of inputs of the transparent bundle is non-empty. It should also be able to verify that the amount credited to the PGF account is in a supported token and matches the expected amount for that token

Reply:
Tiago was proposing the same thing. It would indeed work to put this as an extra data section. It seems really hacky to me, but we can discuss at standup.

Reply:
The difficult part would be knowing which masp tx was paid for it multiple shields are in a batch. We need a commitment.